src/components/sidebar/SafeList/index.tsx

```diff
@@ -63,17 +66,36 @@ const SafeList = ({ closeDrawer }: { closeDrawer: () => void }): ReactElement =>
     setOpen((prev) => ({ ...prev, [chainId]: open }))
   }
 
+  useEffect(() => {
+    const addedSafeAddresses = Object.keys(addedSafes?.[chainId] || {})
+
+    if (addedSafeAddresses.length > 0) {
+      return
+    }
+
+    const event = OVERVIEW_EVENTS.ADDED_SAFES_ON_NETWORK
+    const { chainName } = configs.find((chain) => chain.chainId === chainId) || {}
+
+    trackEvent({
+      ...event,
+      action: `${event.action} ${chainName}`,
+      label: addedSafeAddresses.length,
+    })
+  }, [addedSafes, chainId])
```

Comment:
IMHO not the best place to track it. I would put it to the added safes middleware.